packages/cli/tests/e2e/solution/AddCapabilities_TabAddBot.tests.ts

```diff
@@ -19,31 +19,39 @@ import {
 import "mocha";
 import { CliHelper } from "../../commonlib/cliHelper";
 import { Capability } from "../../commonlib/constants";
-
+import mockedEnv, { RestoreFn } from "mocked-env";
 describe("Add capabilities", function () {
   const testFolder = getTestFolder();
   const subscription = getSubscriptionId();
-  const appName = getUniqueAppName();
-  const projectPath = path.resolve(testFolder, appName);
-  const env = environmentManager.getDefaultEnvName();
-
-  after(async () => {
-    await cleanUp(appName, projectPath, true, true, false);
-  });
-
-  it("tab project can add bot capability and provision", async () => {
-    // Arrange
-    await CliHelper.createProjectWithCapability(appName, testFolder, Capability.Tab);
-
-    // Act
-    await CliHelper.addCapabilityToProject(projectPath, Capability.Bot);
-
-    await setSimpleAuthSkuNameToB1Bicep(projectPath, env);
-    await setBotSkuNameToB1Bicep(projectPath, env);
-    await CliHelper.setSubscription(subscription, projectPath);
-    await CliHelper.provisionProject(projectPath);
-
-    // Assert
-    await validateTabAndBotProjectProvision(projectPath, env);
+  let appName: string | undefined;
+  let projectPath: string | undefined;
+  let mockedEnvRestore: RestoreFn | undefined;
+  afterEach(async () => {
+    if (appName && projectPath) {
+      await cleanUp(appName, projectPath, true, true, false);
+    }
+    if (mockedEnvRestore) {
+      mockedEnvRestore();
+    }
   });
+  for (const v3flag of ["false"]) {
```

Comment:
Do you want to test the flag with "true"?

Reply:
Yes, after fix the getQuestion problem